src/projectM-sdl/projectM_SDL_main.cpp

```diff
@@ -53,28 +53,55 @@ void DebugLog(GLenum source,
 #endif
 
 // return path to config file to use
-std::string getConfigFilePath(std::string datadir_path) {
-    struct stat sb;
-    const char *path = datadir_path.c_str();
-    
-    SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, "Looking for configuration file in data dir: %s.\n", path);
-    
-    // check if datadir exists.
-    // it should exist if this application was installed. if not, assume we're developing it and use currect directory
-    if (stat(path, &sb) != 0) {
-        SDL_LogWarn(SDL_LOG_CATEGORY_ERROR, "Could not open datadir path %s\n", path);
-    }
-    
-    std::string configFilePath = path;
-    configFilePath += "/config.inp";
-    
-    // check if config file exists
-    if (stat(configFilePath.c_str(), &sb) != 0) {
-        SDL_LogWarn(SDL_LOG_CATEGORY_ERROR, "No config file %s found. Using development settings.\n", configFilePath.c_str());
-        return "";
-    }
-    
-    return configFilePath;
+std::string getConfigFilePath(std::string datadir_path)
+{
+	struct stat sb;
+	const char *path = datadir_path.c_str();
+
+	// first check if settings file is in DATA_DIR
+	std::string configFilePath = path;
+	std::string configFullPath = configFilePath;
+	configFullPath += "config.inp";
+
+	if (stat(configFullPath.c_str(), &sb) == 0)
+	{
+		SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION, "Config: %s was found!\n", configFullPath.c_str());
+		return configFilePath;
+	}
+
+	// now check settings directories (windows / other)
+	configFilePath = "";
+#ifdef WIN32
+	configFilePath += getenv("APPDATA");
```

Comment:
what does this return?

Reply:
Previously it would return the full path to the config file (if it exists), or an empty string if it does not exist.

Now it returns the **folder** of the config file (if it exists), or an empty string if does not exist. 

I changed it so the folder where the config file was found can easily be used as the new base_path for built-in config settings like the presets folders.